bin/strata-cli/src/seed/password.rs

```diff
@@ -58,23 +54,17 @@ impl Password {
         &mut self,
         salt: &[u8; PW_SALT_LEN],
         version: HashVersion,
-    ) -> Result<&[u8; 32], argon2::Error> {
-        match self.seed_encryption_key {
-            Some(ref key) => Ok(key),
-            None => {
-                let mut sek = [0u8; 32];
-                let (algo, ver, params) = version.params();
-                if !self.inner.is_empty() {
-                    Argon2::new(algo, ver, params.expect("valid params")).hash_password_into(
-                        self.inner.as_bytes(),
-                        salt,
-                        &mut sek,
-                    )?;
-                }
-                self.seed_encryption_key = Some(sek);
-                self.seed_encryption_key(salt, version)
-            }
+    ) -> Result<[u8; 32], argon2::Error> {
+        let mut sek = [0u8; 32];
+        let (algo, ver, params) = version.params();
+        if !self.inner.is_empty() {
```

Comment:
Why check for an empty string here? A derived key from an empty string, while perhaps something of an edge case, is still valid.

Reply:
I'd pass in the empty byte array anyway, which removes the edge case entirely.